src/java/org/apache/cassandra/index/sai/disk/hnsw/CassandraOnDiskHnsw.java

```diff
@@ -171,44 +169,27 @@ public OnDiskOrdinalsMap.OrdinalsView getOrdinalsView() throws IOException
         return ordinalsMap.getOrdinalsView();
     }
 
-    class OnDiskVectors implements RandomAccessVectorValues<float[]>, AutoCloseable
+    class VectorsWithCache implements RandomAccessVectorValues<float[]>, AutoCloseable
     {
-        private final RandomAccessReader reader;
-        private final long segmentOffset;
-        private final int dimension;
-        private final int size;
-        private final float[] vector;
+        private final OnDiskVectors vectors;
         private final QueryContext queryContext;
 
-        public OnDiskVectors(FileHandle fh, long segmentOffset, QueryContext queryContext)
+        public VectorsWithCache(OnDiskVectors vectors, QueryContext queryContext)
```

Comment:
This is responsible for closing the `OnDiskVectors`, but it doesn't seem to be being closed. I think we need to create distinct `OnDiskVectors` and `VectorsWithCache`  fields and close the `VectorsWithCache` in `close`.

Reply:
Which leak is that? The `vectorsSupplier` still creates a `VectorsWithCache` that isn't closed anywhere. This will still result in a leak.